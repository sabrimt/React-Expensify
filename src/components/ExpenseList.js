import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

// const ExpenseList = (props) => (
//     <div>
//         <h1>Expense List</h1>
//         <p><strong>Sort By: </strong>{props.filters.text}</p>
//         {props.expenses.map((expense) => (<p key={expense.description}>{expense.description}</p>))}
//     </div>
// );
//
// const mapStateToProps = (state) => {
//
//     return {
//         expenses: getVisibleExpenses(state.expenses, state.filters),
//         filters: state.filters
//     };
// };


export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map(
                    (expense) => <ExpenseListItem key={expense.id} {...expense} />
                )
            )
        }
        {}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
