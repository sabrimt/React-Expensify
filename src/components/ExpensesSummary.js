import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <h4>
            viewing {props.expenseCount} expense{props.expenseCount !== 1 && 's'} totalling {
                numeral(props.expensesTotal / 100).format('$0,0.00')
            }
        </h4>
    </div>
);

const mapStateToProps = ((state) => ({
    expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(getVisibleExpenses(state.expenses, state.filters))
}));

export default connect(mapStateToProps)(ExpensesSummary);
