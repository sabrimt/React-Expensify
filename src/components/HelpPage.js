import React from 'react';

const HelpPage = () => (
    <div>
        this is from HELP page
    </div>
);

export default HelpPage;
