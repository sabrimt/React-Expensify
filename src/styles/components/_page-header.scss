.page-header {
    background: $off-white;
    margin-bottom: $l-size;
    padding: $l-size 0;
}

.page-header__actions {
    margin-top: $l-size;
}

.page-header__title {
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 700;
    }
}

.page-header__sub-title {
    color: $grey;
    font-style: italic;
    font-weight: 300;
    margin: 0;
    span {
        font-weight: 600;
    }
}
