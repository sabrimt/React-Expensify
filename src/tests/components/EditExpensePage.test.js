import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { EditExpensePage } from '../../components/EditExpensePage';

let wrapper, startEditExpense, startRemoveExpense, history;

beforeEach(() => {
    startEditExpense = jest.fn();
    startRemoveExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage
        startEditExpense={startEditExpense}
        startRemoveExpense={startRemoveExpense}
        history={history}
        expense={expenses[1]}
    />);
});

test('Should render EditExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle startEditExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(startEditExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1]);
});

test('Should handle startRemoveExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(startRemoveExpense).toHaveBeenLastCalledWith({ id: expenses[1].id });
});
