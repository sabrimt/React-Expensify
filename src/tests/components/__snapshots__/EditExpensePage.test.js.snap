// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 70000,
        "createdAt": -345600000,
        "description": "PC bill",
        "id": "2",
        "note": "info",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
